### ISSUE TITLE
can someone help with installing cv2.

### ISSUE THREAD

**Issue by vivek-ram** [apple/tensorflow_macos#79]
hi i am using conda environment i need to installcv2 is there any why i can install cv2 ??

**Comment by MintStudios** [comment]
Duplicate of #67

**Comment by bjajoh** [comment]
got opencv and conda working correctly. I documented my procedure here and will add more in the future.
https://github.com/bjajoh/cv2-tf-scipy-for-arm-mac/blob/main/README.md

This is what I came up with:

```shell
cmake -DCMAKE_BUILD_TYPE=Release \
-DBUILD_opencv_python2=OFF \
-DBUILD_opencv_python3=ON \
-DPYTHON3_EXECUTABLE=`which python3` \
-DPYTHON3_INCLUDE_DIR=/Users/USER/miniforge3/envs/python38/include/python3.8 \
-DPYTHON_INCLUDE=/Users/USER/miniforge3/envs/python38/include/python3.8 \
-DPYTHON3_PACKAGES_PATH=$(python -c "import sysconfig; print(sysconfig.get_config_var('LIBDEST'))") \
-DPYTHON3_NUMPY_INCLUDE_DIR=/Users/USER/miniforge3/envs/python38/lib/python3.8/site-packages/numpy/core \
-DCMAKE_INSTALL_PREFIX=/usr/local \
-DOPENCV_EXTRA_MODULES_PATH=~/opencv_contrib/modules \
-DOPENCV_ENABLE_NONFREE=ON \
../opencv
```

**Comment by vivek-ram** [comment]
@bjajoh 'cmake -DCMAKE_BUILD_TYPE=Release \' where implement this ?? i mean how to do this. should we create a cmakelist.txt file in build_opencv directory ? or what

**Comment by bjajoh** [comment]
@vivek-ram The last line shows the location. When following my guide, there should be a source folder with OpenCV in your home directory. Furthermore, make sure to install CMake. I installed it from source.
```shell
https://github.com/Kitware/CMake.git
./bootstrap && make && sudo make install
```
Here is the full guide: https://github.com/bjajoh/cv2-tf-scipy-for-arm-mac/blob/main/README.md

**Comment by vivek-ram** [comment]
@bjajoh i copied and pasted the cmake thing but i an error.
<img width="723" alt="Screenshot 2020-12-23 at 9 27 54 pm" src="https://user-images.githubusercontent.com/56974835/103038190-bf64cd80-4565-11eb-9bdb-b467e06aa067.png">

**Comment by bjajoh** [comment]
Interesting, I documented everything after I finally managed to get everything working, so there is the risk that I missed a small detail. Is there a more detailed error message? Did you clone both repositories correctly in your home folder? So "opencv_contrib" and "opencv"?

**Comment by vivek-ram** [comment]
@bjajoh  yeah i did.

**Comment by bjajoh** [comment]
It is hard to say what is wrong without any specific error. Maybe check the output of that cmake command and try to see if anything there seems wrong.

**Comment by vivek-ram** [comment]
@bjajoh nothing seems wrong bro. 

cmake -DCMAKE_BUILD_TYPE=Release \
-DBUILD_opencv_python2=OFF \
-DBUILD_opencv_python3=ON \
-DPYTHON3_EXECUTABLE=`which python3` \
-DPYTHON3_INCLUDE_DIR=/Users/vivekd/miniforge3/envs/python38/include/python3.8 \
-DPYTHON_INCLUDE=/Users/vivekd/miniforge3/envs/python38/include/python3.8 \
-DPYTHON3_PACKAGES_PATH=$(python -c "import sysconfig; print(sysconfig.get_config_var('LIBDEST'))") \
-DPYTHON3_NUMPY_INCLUDE_DIR=/Users/vivekd/miniforge3/envs/python38/lib/python3.8/site-packages/numpy/core \
-DCMAKE_INSTALL_PREFIX=/usr/local \
-DOPENCV_EXTRA_MODULES_PATH=/Users/vivekd/opencv_contrib/modules \
-DOPENCV_ENABLE_NONFREE=ON \
../opencv


this is what i used. i followed your steps and i copied this in terminal it was good. next when i run make j7 getting that error 2

**Comment by sanmeow** [comment]
> @bjajoh nothing seems wrong bro.
> 
> cmake -DCMAKE_BUILD_TYPE=Release 
> -DBUILD_opencv_python2=OFF 
> -DBUILD_opencv_python3=ON 
> -DPYTHON3_EXECUTABLE=`which python3` 
> -DPYTHON3_INCLUDE_DIR=/Users/vivekd/miniforge3/envs/python38/include/python3.8 
> -DPYTHON_INCLUDE=/Users/vivekd/miniforge3/envs/python38/include/python3.8 
> -DPYTHON3_PACKAGES_PATH=$(python -c "import sysconfig; print(sysconfig.get_config_var('LIBDEST'))") 
> -DPYTHON3_NUMPY_INCLUDE_DIR=/Users/vivekd/miniforge3/envs/python38/lib/python3.8/site-packages/numpy/core 
> -DCMAKE_INSTALL_PREFIX=/usr/local 
> -DOPENCV_EXTRA_MODULES_PATH=/Users/vivekd/opencv_contrib/modules 
> -DOPENCV_ENABLE_NONFREE=ON 
> ../opencv
> 
> this is what i used. i followed your steps and i copied this in terminal it was good. next when i run make j7 getting that error 2

I faced the same problem. Solved by using this article's way.
https://sayak.dev/install-opencv-m1
I still faced the same error 2.
Some other parts trying, don't know which one made the compile finish.
1.Don't build inside using virtual environment.
2. Update Eigen to 3.3.9
Really happy it got solved.

**Comment by isuruf** [comment]
You can now install using conda `conda install opencv`

**Comment by vivek-ram** [comment]
@isuruf wow thanks bro finally solved.

**Comment by bjajoh** [comment]
Yeah, that's awesome! But it could still be an issue if you want to compile the dnn stuff of opencv 4 together with the accelerated Tensorflow. Conda only works out if you're not using the DNN Features of opencv.

**Comment by isuruf** [comment]
@bjajoh, help is welcome at https://github.com/conda-forge/opencv-feedstock
